File: 88-MergeSortedArray/88-MergeSortedArray.cpp
// Last updated: 08/05/2026, 09:32:38
class Solution {
public:

    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int k= m+n;
        int j1=0;
        int j2=0;

        for(int i=0;i<k; i++){
            if(j1<m && j2<n){
            if(nums1[m-1-j1]>nums2[n-1-j2]){
                nums1[k-1-i]=nums1[m-1-j1];
                j1+=1;}
            
            else if(nums1[m-1-j1]<=nums2[n-1-j2]){
                nums1[k-1-i]=nums2[n-1-j2];
                j2+=1;
            }
            }
            else if(j1<m){
                nums1[k-1-i]= nums1[m-1-j1];
                j1+=1;
            }
            else if(j2<n){
                nums1[k-1-i]= nums2[n-1-j2];
                j2+=1;
            }   
    }}

};

File: 88-MergeSortedArray/88-MergeSortedArray.py
# Last updated: 08/05/2026, 09:32:33
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """

        left= m-1
        e= m+n-1
        i=n-1

        while i>=0 and left>=0:
            if nums1[left]<nums2[i]:
                nums1[e]= nums2[i]
                i-=1
            
            else:
                nums1[e]= nums1[left] 
                left-=1
            
            e-=1

        while i>=0:
            nums1[e]=nums2[i]
            e-=1
            i-=1

        